### PR TITLE
task/SC-1100/node_js_12_deprecation

### PR DESCRIPTION
**Pull request by Ezeigwefabian**
## Jira Story
SC-1100

## About

1. actions/checkout@v2 -> actions/checkout@v3
2. actions/setup-python@v1 -> actions/setup-python@v4
3. aws-actions/configure-aws-credentials@v1 -> aws-actions/configure-aws-credentials@v1-node16
4. actions/upload-artifact@v2 ->v3
5. #Install Poetry
          python -m ensurepip
          python -m pip install --upgrade pip
          python -m pip install poetry==1.1.15 -->  python3 -m ensurepip
          python3 -m pip install --upgrade pip
          python3 -m pip install poetry==1.1.15 SecretStorage==3.3.3 cachecontrol==0.12.11 cachy==0.3.0 certifi==2022.12.7 cffi==1.15.1 charset-normalizer==3.1.0 cleo==0.8.1 clikit==0.6.2 crashtest==0.3.1 cryptography==40.0.2 distlib==0.3.6 filelock==3.12.0 html5lib==1.1 idna==3.4 importlib-metadata==1.7.0 jeepney==0.8.0 keyring==22.3.0 lockfile==0.12.2 msgpack==1.0.5 packaging==20.9 pastel==0.2.1 pexpect==4.8.0 pkginfo==1.9.6 platformdirs==2.6.2 poetry-core==1.0.8 ptyprocess==0.7.0 pycparser==2.21 pylev==1.4.0 pyparsing==3.0.9 requests==2.29.0 requests-toolbelt==0.9.1 shellingham==1.5.0.post1 six==1.16.0 tomlkit==0.11.8 typing-extensions==4.5.0 urllib3==1.26.15 virtualenv==20.16.2 webencodings==0.5.1 zipp==3.15.0
          poetry --version
          poetry config virtualenvs.in-project true
          poetry config virtualenvs.path .

## Associated Pull Requests
N/A

## Update Changelog

*   [X] Changelog updated

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -26,6 +26,7 @@ For instructions, see the [changelog confluence page](https://epcpower.atlassian
 - SC-760: Pin poetry to 1.1.15 & fix boto3 ref
 - SC-399: Romp Removal / Poetry Implementation
 - SC-219: Github Actions CI
+- SC-1100: Update actions versions to alleviate CI build warnings
```
Comment (gordon-epc) [inline CHANGELOG.md:29]: Move this row to the top CI line (just above SC-798: ...)

**Review by gordon-epc — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml
Comment (gordon-epc) [inline .github/workflows/ci.yml:1]: This line needs to be updated:
`uses: actions/upload-artifact@v2`
which is giving this warning message:
```
Node.js 12 actions are deprecated. Please update the following actions to use Node.js 16: actions/upload-artifact@v2. For more information see: ...
```

**Review by gordon-epc — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml
Comment (Fabian6781) [reply .github/workflows/ci.yml:1]: I guess node16 shouldn't be there, i took it out and the build succeeded.

Comment (gordon-epc) [inline .github/workflows/ci.yml:1]: Still has [the warning](https://github.com/epcpower/stlib/actions/runs/4694242793):

`Node.js 12 actions are deprecated. Please update the following actions to use Node.js 16: actions/upload-artifact@v2. For more information see: https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/.`

You need to update:
`actions/upload-artifact@v2`

**Review by gordon-epc — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -22,6 +22,8 @@ For instructions, see the [changelog confluence page](https://epcpower.atlassian
 
 ### CI
 
+- SC-1100: Update ci.yml actions to fix poetry version issue
```
Comment (gordon-epc) [inline CHANGELOG.md:25]:
```suggestion
- SC-1100: Pin versions of all python packages for installing poetry
```

**Review by gordon-epc — CHANGES_REQUESTED** [review]

**Review by gordon-epc — APPROVED** [review]

**Pull request merged**